### PR TITLE
Add test for from_address fixes for activation email 

### PR DESCRIPTION
**Pull request by burhandodhy**
This PR add test to the fixes of from_address. The only way was to assert the `from_address` from  the email error logs because `compose_and_send_activation_email` doesn't return or log any message on email success. 

Relevant PR https://github.com/edly-io/edx-platform/pull/228

### CODE REVIEW

**Review by asadmanzoor93 — APPROVED** [review]

File: common/djangoapps/student/tests/test_tasks.py [hunk 1]
```diff
@@ -91,3 +92,27 @@ def test_UnrecoverableSendError(self, mock_log):
...
         self.assertEqual(mock_log.error.call_count, 0)
         self.assertEqual(mock_log.exception.call_count, 1)
+
+    @mock.patch('student.tasks.log')
+    @mock.patch('student.tasks.ace.send', mock.Mock(side_effect=ChannelError))
+    @mock.patch('common.djangoapps.student.views.management.theming_helpers.get_current_site')
+    @mock.patch('openedx.core.djangoapps.site_configuration.helpers.get_current_site_configuration')
+    def test_from_address_in_send_email(self, mock_site_configuration, mock_get_current_site, mock_log):
+        """
+        Tests that a from address pull from site configuration.
```
Comment (zubair-arbi) [inline common/djangoapps/student/tests/test_tasks.py:102]: `Tests that the "from_address" is pulled from the site configuration.`

File: common/djangoapps/student/tests/test_tasks.py [hunk 2]
```diff
@@ -91,3 +92,27 @@ def test_UnrecoverableSendError(self, mock_log):
...
+        """
+        site = SiteFactory.create()
+        mock_get_current_site.return_value = site
+        from_address = 'no-reply@example.com'
```
Comment (zubair-arbi) [inline common/djangoapps/student/tests/test_tasks.py:106]: We usually use `no-reply@example.com` email address as default value. I would suggest to use `test-no-reply@example.com` for testing.

```python
expected_from_email_address = 'test-no-reply@example.com'
```

File: common/djangoapps/student/tests/test_tasks.py [hunk 3]
```diff
@@ -91,3 +92,27 @@ def test_UnrecoverableSendError(self, mock_log):
...
+        site_config = SiteConfigurationFactory.create(site=site, site_values={
+            'ACTIVATION_EMAIL_FROM_ADDRESS': from_address
+        })
+        mock_site_configuration.return_value = site_config
+
+        compose_and_send_activation_email(self.student, self.student.profile, self.student.registration)
+        # Asserts that the from_address logged in error message.
```
Comment (zubair-arbi) [inline common/djangoapps/student/tests/test_tasks.py:113]: remove this comment

File: common/djangoapps/student/tests/test_tasks.py [hunk 4]
```diff
@@ -9,10 +9,11 @@
 from six.moves import range
 
 from edx_ace.errors import ChannelError, RecoverableChannelDeliveryError
+from openedx.core.djangoapps.site_configuration.tests.factories import SiteConfigurationFactory, SiteFactory
```
Comment (zubair-arbi) [inline common/djangoapps/student/tests/test_tasks.py:12]: use alphabetical ordering in imports

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
I have added my feedback

File: common/djangoapps/student/tests/test_tasks.py (hunk 1 above)
Comment (burhandodhy) [reply common/djangoapps/student/tests/test_tasks.py]: Done

File: common/djangoapps/student/tests/test_tasks.py (hunk 2 above)
Comment (burhandodhy) [reply common/djangoapps/student/tests/test_tasks.py]: Done

File: common/djangoapps/student/tests/test_tasks.py (hunk 3 above)
Comment (burhandodhy) [reply common/djangoapps/student/tests/test_tasks.py]: Done

File: common/djangoapps/student/tests/test_tasks.py (hunk 4 above)
Comment (burhandodhy) [reply common/djangoapps/student/tests/test_tasks.py]: Done

**Review by zubair-arbi — APPROVED** [review]
Changes looks good 👍

**Pull request merged**